src/sortStudents.ts

```diff
@@ -1,16 +1,97 @@
 
 export interface Student {
-  // describe Student interface
+  name: string,
+  surname: string,
+  age: number,
+  married: boolean,
+  grades: number[],
 }
 
 export enum SortType {
-  // describe SortType enum
+  Name = 'name',
+  Surname = 'surname',
+  Age = 'age',
+  Married = 'married',
+  AverageGrade = 'grades'
 }
 
-// create SortOrder type
-export type SortOrder;
+export type SortOrder = 'asc' | 'desc';
 
+function getAverageGrade(grades: number[]): number {
+  return grades.reduce((sum, current) => sum + current, 0) / grades.length;
+}
+
+function sortByAverageGrade(
+  students: Student[],
+  order: string,
+): Student[] {
+  if (order === 'asc') {
+    return students
+      .sort((prev, next) => getAverageGrade(prev.grades)
+        - getAverageGrade(next.grades));
```

Comment:
```suggestion
    return students.sort((prev, next) => 
      getAverageGrade(prev.grades) - getAverageGrade(next.grades)
    );
```
but I would be nice if you create a callback to make it look decomposed